mindmeld/augmentation.py

```diff
@@ -0,0 +1,324 @@
+            )
+            return []
+
+        rules = paths
+
+        filtered_paths = []
+
+        for rule in rules:
+            pattern = get_pattern(rule)
+            compiled_pattern = re.compile(pattern)
+            filtered_paths.extend(
+                self._resource_loader.filter_file_paths(
+                    compiled_pattern=compiled_pattern, file_paths=all_file_paths
+                )
+            )
+        return filtered_paths
+
+    def _read_path_queries(self, path):
+        """Returns all queries in a specified file path.
+
+        Args:
+            path (str): File path.
+
+        Return:
+            queries (list): List of queries in the file path.
+        """
+        with open(path, "r") as f:
+            queries = f.readlines()
+        return queries
+
+    def _write_files(self, path, augmented_queries, suffix):
+        """Writes augmented queries to a new file in the path.
+
+        Args:
+            path (str): File path to the original file.
+            augmented_queries (list): List of augmented queries returned by class.
+        """
+        write_path = path.strip(".txt") + suffix
+
+        with open(write_path, "w") as outfile:
+            for query in augmented_queries:
+                outfile.write(query.strip("\n") + "\n")
+
+
+class EnglishParaphraser(Augmentor):
+    """Paraphraser class for generating English paraphrases."""
+
+    def __init__(self, lang, paths, path_suffix, params, resource_loader):
+        """Initializes an English paraphraser.
+
+        Args:
+            lang (str): The lang code for paraphrasing
+            paths (list): Path rules for fetching relevant files to Paraphrase.
+            path_suffix (str): Suffix to be added to new augmented files.
+            params (dict): Model params.
+            resource_loader (object): Resource Loader object for the application.
+        """
+        super().__init__(
+            lang=lang,
+            paths=paths,
+            path_suffix=path_suffix,
+            resource_loader=resource_loader,
+        )
+
+        PegasusTokenizer = _get_module_or_attr("transformers", "PegasusTokenizer")
+        PegasusForConditionalGeneration = _get_module_or_attr(
+            "transformers", "PegasusForConditionalGeneration"
+        )
+
+        model_name = "tuner007/pegasus_paraphrase"
+        torch_device = (
+            "cuda" if _get_module_or_attr("torch.cuda", "is_available")() else "cpu"
+        )
+        self.tokenizer = PegasusTokenizer.from_pretrained(model_name)
+        self.model = PegasusForConditionalGeneration.from_pretrained(
+            model_name, force_download=False
+        ).to(torch_device)
+
+        # Update default params with user model config
+        self.params = {
+            "max_length": 60,
+            "num_beams": 10,
+            "num_return_sequences": 10,
+            "temperature": 1.5,
+        }
+
+        if params:
+            self.params.update(params)
```

Comment:
As discussed, lets try and get rid of params and instead enumerate the primitive args for the model in the constructor with good defaults.